src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -432,6 +410,43 @@ bool IsRemovableAssignmentWithoutSideEffects(IOperation unusedSymbolWriteOperati
                         // Assume all other operations can have side effects, and cannot be removed.
                         return false;
                     }
+
+                    bool IsRemovableAssignmentValueWithoutSideEffects(IOperation assignmentValue)
+                    {
+                        if (assignmentValue.ConstantValue.HasValue)
+                        {
+                            // Constant expressions have no side effects.
+                            return true;
+                        }
+
+                        switch (assignmentValue.Kind)
+                        {
+                            case OperationKind.ParameterReference:
+                            case OperationKind.LocalReference:
+                                // Parameter/local references have no side effects and can be removed.
+                                return true;
+
+                            case OperationKind.FieldReference:
+                                // Field references with null instance (static fields) or 'this' or 'Me' instance can
+                                // have no side effects and can be removed.
+                                var fieldReference = (IFieldReferenceOperation)assignmentValue;
+                                return fieldReference.Instance == null || fieldReference.Instance.Kind == OperationKind.InstanceReference;
+
+                            case OperationKind.DefaultValue:
+                                // Default value expressions have no side-effects.
+                                return true;
+
+                            case OperationKind.Conversion:
+                                // Conversions can theoretically have side-effects as the conversion can throw exception(s).
+                                // However, for all practical purposes, we can assume that a conversion whose operand
+                                // has no side effects can be safely removed.
```

Comment:
thanks for the comment.  note: i would be a little more conservative and just do this for conversions that do *not* involve a user-defined-method.

Reply:
Done with https://github.com/dotnet/roslyn/pull/30777/commits/10944f355ad17bce1695f53afad3c6a3895bc357